frontend/src/pages/modelServing/utils.ts

```diff
@@ -160,7 +160,15 @@ export const createRoleBindingIfMissing = async (
 ): Promise<RoleBindingKind> =>
   getRoleBinding(namespace, rolebinding.metadata.name).catch((e) => {
     if (e.statusObject?.code === 404) {
-      return createRoleBinding(rolebinding, opts);
+      return createRoleBinding(rolebinding, opts).catch((error) => {
+        if (error.statusObject?.code === 404 && opts?.dryRun) {
+          // If dryRun is enabled and the user is not Cluster Admin it seems that there's a k8s error
+          // that raises a 404 trying to find the role, which is missing from the dryRun
```

Comment:
Missing from the Dry Run -- because we are dry running everything and the RoleBinding validates the existence of the Role on the cluster?

Reply:
Yup, that's right, I assume this is a k8s error, cause it only happens with permisionless users, when dryRunning, creating the rolebinding i think internally tries to get the role it's trying to assign, since it's not yet created it fails. My guess is that with cluster admins k8s does the correct check and it assumes it's dryRunned too.

Reply:
It's funny, cause if you change the rolebinding to not use an specific role it works just fine, or if you run this with cluster admins, it's just with this specific scenario.

Reply:
This is definitely strange. It doesn't make sense that the `createRoleBinding` util would return a 404 since it's a direct call to `k8sCreateResource` with the `RoleBindingModel`, the client never does a GET for the role - you mean that is happening internally in the k8s API? Seems bizarre that it would respond with 404 then and not 400.

It also doesn't make sense that this ever worked for admins tbh - I don't see why the API would be aware of the previous dryRun requests, they're all separate HTTP requests. So for some reason that internal check for the role existing isn't happening for admins? I wonder if admin permissions mean overriding checks like that or something? Confusing all around...

This solution seems fine to me based on the behavior we're seeing, I would love to understand it more but I suppose that doesn't block this PR.